app/src/main/java/com/simplecity/amp_library/model/AlbumArtist.java

```diff
@@ -114,9 +114,7 @@ public boolean equals(Object o) {
 
         AlbumArtist that = (AlbumArtist) o;
 
-        if (name != null ? !name.equals(that.name) : that.name != null) return false;
-        return albums != null ? albums.equals(that.albums) : that.albums == null;
-
+        return name != null ? name.equals(that.name) : that.name == null && (albums != null ? albums.equals(that.albums) : that.albums == null);
```

Comment:
I'd prefer not to make this change. 'Simplifying' the conditional statement generally makes it harder to read for equals() method.